### PR TITLE
Refactor/Beta::ProjectBoardOperations::Read

### PR DESCRIPTION
**Pull request by igor-dmscn**
Refactor Beta::ProjectBoardOperations to use [dry-monads](https://github.com/dry-rb/dry-monads) and [dry-matcher](https://github.com/dry-rb/dry-matcher), making use of variables and calls clearer.

### CODE REVIEW

File: app/operations/beta/project_board_operations/read.rb [hunk 1]
```diff
@@ -1,56 +1,64 @@
...
+        @project_id = project_id
         @current_user = current_user
         @current_flow = current_flow
         @projects_scope = projects_scope
       end
 
-      def run
-        users = project.users
+      def call
+        ActiveRecord::Base.transaction do
```
[the remark this answers is not in the document]
Comment (igor-dmscn) [reply app/operations/beta/project_board_operations/read.rb:14]: didn't realize that. done

File: app/operations/beta/project_board_operations/read.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+      def stories_and_past_iterations
+        read_all_result = yield ::StoryOperations::ReadAll.call(project: project)
+        read_all_result[:stories] = read_all_result.delete(:active_stories)
+        read_all_result
+      end
+
+      def project_labels
+        labels = project.stories.map(&:labels).reject { |label| label.to_s.empty? }
+        labels.join(',').split(',').uniq.join(',')
```
Comment (nolleto) [inline app/operations/beta/project_board_operations/read.rb:50]: Why are you `join` and then `split` here?

It seems you don't need that 🤔 

```diff
- labels.join(',').split(',').uniq.join(',')
+ labels.uniq.join(',')
```

Another thing, you may write like this:

```ruby
def project_labels
  project.stories
         .map(&:labels)
         .reject { |label| label.to_s.empty? }
         .uniq
         .join(',')
end
```

Comment (igor-dmscn) [reply app/operations/beta/project_board_operations/read.rb:50]: actually Idk. didn't change the previous code, but I'll try

File: app/operations/beta/project_board_operations/read.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      end
+
+      def stories_and_past_iterations
+        read_all_result = yield ::StoryOperations::ReadAll.call(project: project)
+        read_all_result[:stories] = read_all_result.delete(:active_stories)
+        read_all_result
+      end
+
+      def project_labels
+        labels = project.stories.map(&:labels).reject { |label| label.to_s.empty? }
```
[the remark this answers is not in the document]
Comment (igor-dmscn) [reply app/operations/beta/project_board_operations/read.rb:49]: makes sense. changed and it worked

File: app/operations/beta/project_board_operations/read.rb (hunk 2 above)
Comment (igor-dmscn) [reply app/operations/beta/project_board_operations/read.rb:50]: I see the problem now.
each entry of label can be a set of labels, so the `.map` return something like `["front,back", "front", "bug,back"]`
that's why it is using the "double" `join`.

Comment (igor-dmscn) [reply app/operations/beta/project_board_operations/read.rb]: changed variable names to address this

**Pull request merged**